### PR TITLE
Using stub data for features check

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

**Comment by robertbastian** [comment]
That's a bingo!

File: tools/make/tests.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ description = "Build all permutations of all features for the provided chunk ind
 category = "ICU4X Development"
 install_crate = { crate_name = "cargo-all-features", binary = "cargo-check-all-features", test_arg = ["--help"] }
 install_crate_args = ["--version", "^1.10"]
-env = { RUSTFLAGS = "-Dwarnings" }
+env = { RUSTFLAGS = "-Dwarnings",  "ICU4X_DATA_DIR" = "${CARGO_MAKE_WORKSPACE_WORKING_DIRECTORY}/provider/datagen/tests/data/baked" }
```
Comment (eggrobin) [inline tools/make/tests.toml:12]: Is this going to work on Windows? Is it covered by the Windows CI, or alternatively is there some command I can try running to check if it does?

**Comment by robertbastian** [comment]
Should work, try `cargo make quick`

**Review by eggrobin — APPROVED** [review]
Seems to work on Windows indeed, for some definition of « quick » :
```
> cargo make quick
[…]
[cargo-make] INFO - Build Done in 496.08 seconds.
```

**Comment by sffc** [comment]
> Seems to work on Windows indeed, for some definition of « quick » :
> 
> ```
> > cargo make quick
> […]
> [cargo-make] INFO - Build Done in 496.08 seconds.
> ```

Subsequent runs should be quicker, too

**Review by sffc — COMMENTED** [review]
Nice! Love it when CI gets faster. Also good catch about adding `-Dwarnings`.

**Comment by Manishearth** [comment]
Are we okay with only using stubdata on features CI? I guess as long as the real data is used for the main test CI we're fine.

(I know @sffc is more conservative than I am when it comes to ensuring full CI coverage: I'm more willing to cut corners for CI speed, so given that he's okay with this this time I am more easily convinced this is fine)

**Comment by robertbastian** [comment]
Full baked data is still tested at MSRV even, in the msrv-check job. It's just the features permutations that use stub data.

**Comment by sffc** [comment]
Using stubdata on features CI is good with me; we use full data in other CIs.

**Comment by sffc** [comment]
Also, stubdata includes all data keys and at least one entry for each key (`und` and `und-*`), so if you can build stubdata your chances of being able to build full data are substantially increased.

**Pull request merged**